src/js/utils.mjs

```diff
@@ -42,31 +44,26 @@ clear = false) {
   parentElement.insertAdjacentHTML(position, htmlStrings.join(""));
 }
 
-export function renderWithTemplate(template, 
-  parentElement,
-  data,
-  callback){
-    parentElement.insertAdjacentHTML("afterbegin", template);
-    if (callback) {
-      callback(data);
-    }
+export function renderWithTemplate(template, parentElement, data, callback) {
+  parentElement.insertAdjacentHTML("afterbegin", template);
+  if (callback) {
+    callback(data);
   }
+}
 
-  export async function loadTemplate(path) {
-    const html = await fetch(path).then(convertToText);
-    const template = document.createElement('template');
-    template.innerHTML = html;
-    return template;
-  }
+export async function loadTemplate(path) {
+  const res = await fetch(path);
+  const template = await res.text();
```

Comment:
This was interesting. We needed `template` to be text, but this was returning a DOM element which caused issues.